pkg/cmd/run/watch/watch_test.go

```diff
@@ -0,0 +1,321 @@
+				assert.Error(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+
+			assert.Equal(t, tt.wants.RunID, gotOpts.RunID)
+			assert.Equal(t, tt.wants.Prompt, gotOpts.Prompt)
+			assert.Equal(t, tt.wants.ExitStatus, gotOpts.ExitStatus)
+			assert.Equal(t, tt.wants.Interval, gotOpts.Interval)
+		})
+	}
+}
+
+func TestWatchRun(t *testing.T) {
+	failedRunStubs := func(reg *httpmock.Registry) {
+		inProgressRun := shared.TestRun("more runs", 2, shared.InProgress, "")
+		completedRun := shared.TestRun("more runs", 2, shared.Completed, shared.Failure)
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs"),
+			httpmock.JSONResponse(shared.RunsPayload{
+				WorkflowRuns: []shared.Run{
+					shared.TestRun("run", 1, shared.InProgress, ""),
+					inProgressRun,
+				},
+			}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs/2"),
+			httpmock.JSONResponse(inProgressRun))
+		reg.Register(
+			httpmock.REST("GET", "runs/2/jobs"),
+			httpmock.JSONResponse(shared.JobsPayload{
+				Jobs: []shared.Job{}}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs/2"),
+			httpmock.JSONResponse(completedRun))
+		reg.Register(
+			httpmock.REST("GET", "runs/2/jobs"),
+			httpmock.JSONResponse(shared.JobsPayload{
+				Jobs: []shared.Job{
+					shared.FailedJob,
+				},
+			}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/check-runs/20/annotations"),
+			httpmock.JSONResponse(shared.FailedJobAnnotations))
+	}
+	successfulRunStubs := func(reg *httpmock.Registry) {
+		inProgressRun := shared.TestRun("more runs", 2, shared.InProgress, "")
+		completedRun := shared.TestRun("more runs", 2, shared.Completed, shared.Success)
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs"),
+			httpmock.JSONResponse(shared.RunsPayload{
+				WorkflowRuns: []shared.Run{
+					shared.TestRun("run", 1, shared.InProgress, ""),
+					inProgressRun,
+				},
+			}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs/2"),
+			httpmock.JSONResponse(inProgressRun))
+		reg.Register(
+			httpmock.REST("GET", "runs/2/jobs"),
+			httpmock.JSONResponse(shared.JobsPayload{
+				Jobs: []shared.Job{
+					shared.SuccessfulJob,
+				},
+			}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/check-runs/10/annotations"),
+			httpmock.JSONResponse([]shared.Annotation{}))
+		reg.Register(
+			httpmock.REST("GET", "repos/OWNER/REPO/actions/runs/2"),
+			httpmock.JSONResponse(completedRun))
+		reg.Register(
+			httpmock.REST("GET", "runs/2/jobs"),
+			httpmock.JSONResponse(shared.JobsPayload{
+				Jobs: []shared.Job{
+					shared.SuccessfulJob,
+				},
+			}))
+	}
+
+	tests := []struct {
+		name        string
+		httpStubs   func(*httpmock.Registry)
+		askStubs    func(*prompt.AskStubber)
+		opts        *WatchOptions
+		tty         bool
+		wantErr     bool
+		errMsg      string
+		wantOut     string
+		onlyWindows bool
+		skipWindows bool
+	}{
+		// TODO exit status respected
```

Comment:
Is this to be addressed before shipping?

Reply:
ah yeah it's stray